### ISSUE TITLE
Profile site performance

### ISSUE THREAD

**Issue by toolness** [mozilla/teach.mozilla.org#659]
I'm not actually very familiar with site profiling performance. We should do it! Right now we're following a lot of industry best practices by minifying our JS and CSS for production (and [staging](https://github.com/mozilla/teach.webmaker.org/issues/587#issuecomment-91544597)), as well as offering retina/non-retina versions of images, and a number of [other](https://github.com/mozilla/teach.webmaker.org/pull/448) things, but I'm sure there's always more we can do to speed up the site.

Notes:

* Because login redirects users to id.webmaker.org, it's probably suboptimal for both sites to have their own totally independent bundles. For instance, it means that the user's browser is actually downloading React *twice*. At least for big libraries like that, it might be more useful to fetch them from a CDN.
* ~~It's quite possible that Webpack is including multiple versions of the same library in our bundles. Unfortunately, when I last tried using Webpack's experimental [Deduplication plugin](https://github.com/webpack/docs/wiki/optimization#deduplication) a [few weeks ago](https://github.com/mozilla/teach.webmaker.org/commit/71ee7e4cb7237bca0d0250adbe2c74550f3682aa), it crashed our build process, so I had to leave it out. :(~~ **Update**: On April 24 I was able to upgrade to webpack 1.8.9 and use the Deduplication Plugin without any problems, but unfortunately it didn't actually reduce the size of our JS bundles one bit. :(
* We could potentially lazy-load (or *partially*-load) Tabzilla to avoid loading jQuery all the time (#658).

**Comment by hannahkane** [comment]
Agreed we should do it! Let's consider this a candidate for an upcoming heartbeat.

**Comment by toolness** [comment]
I used [YSlow](http://yslow.org/) on https://teach.mofostaging.net/ and we got a "B", which isn't bad!

All the recommended performance improvements, like [adding cache-control headers](https://developer.yahoo.com/performance/rules.html#expires), seem to be things that we can only really control via S3/CloudFront, and how we do them will change if/when we switch to [using a lightweight server to render HTML](https://github.com/mozilla/teach.webmaker.org/issues/585).

**Comment by toolness** [comment]
I peeked into the modules we're including in our bundles by using Chrome devtools' "Sources" tab, and it looks like there's actually not very much redundancy:

* `react-router` uses [qs](https://www.npmjs.com/package/qs) which is remarkably similar to node's built-in `querystring` module (which we also use).
* `react-select` uses [lodash](https://lodash.com/), while all our other code uses underscore, which means that both of these libraries are included. I guess if we switched to lodash entirely and then enabled webpack's Deduplication plugin, some space could be saved.
* `superagent` uses [component-emitter](https://www.npmjs.com/package/component-emitter) which is a whole lot like node's built-in `EventEmitter` (which we also use).
